haas/api.py

```diff
@@ -510,6 +510,19 @@ def headnode_detach_network(headnode, hnic):
                             # Network Code #
                             ################
 
+@rest_call('GET', '/networks')
+def list_networks():
+    """List all networks.
+    Returns a JSON array of strings representing a list of projects.
+    Example:  {"channels": ["101", "107"], "name": ["net-05", "qnet-00"]}
+    """
+    db = model.Session()
+    networks = db.query(model.Network).all()
+    result = {
+        'channels': [n.network_id for n in networks],
+        'names': [n.label for n in networks],
```

Comment:
Similar comment as to what I posted in #504, I think it would be cleaner for clients to parse to return a single list of either tuples or a dict, where channels are paired with their name, than the current set of two lists.

Also, we should return the project of each network, since this could only presumably be run by admins. It would be nice if we could rename "access" to "project", and if there is a project to output it but if not (ie- it's NULL) to output something like "PUBLIC".

As a dict, this would look something like:
```python
results = {}
for n in networks:
  net = {'channel':n.network_id, 'project':n.project}
  results[n.label] = net

return json.dumps(results)
```